pkg/updater/download.go

```diff
@@ -116,41 +193,142 @@ func extractTarGz(archivePath string, destinationPath string) error {
 		if err != nil {
 			return fmt.Errorf("could not read tar header: %w", err)
 		}
+
+		// Skip the root directory
+		if header.Name == "./" {
+			continue
+		}
+
 		target := filepath.Join(destinationPath, header.Name)
+
+		// Check for zip-slip attacks
+		if !strings.HasPrefix(target, filepath.Clean(destinationPath)+string(os.PathSeparator)) {
+			return fmt.Errorf("tar entry %s is trying to escape the destination directory", header.Name)
+		}
```

Comment:
It's an env var setup in GODEBUG, not sure we can add it here. Also, we have non-local paths sometimes and we don't want to fail the unpack because of them - cf Arthur's message [here](https://github.com/DataDog/datadog-agent/pull/22812#discussion_r1494325600)